Makefile

```diff
@@ -308,7 +306,35 @@ foss-checks: vendor
 # Unit Tests
 ###############################################################################
 ## Run the unit tests.
-ut: run-k8s-controller build $(BIN)/host-local
+ut:
+	make ut-etcd
+	make ut-kdd
+
+ut-etcd: run-k8s-controller build $(BIN)/host-local
+	# The tests need to run as root
+	docker run --rm -t --privileged --net=host \
+	-e ETCD_IP=$(LOCAL_IP_ENV) \
+	-e LOCAL_USER_ID=0 \
+	-e ARCH=$(ARCH) \
+	-e PLUGIN=calico \
+	-e BIN=/go/src/$(PACKAGE_NAME)/$(BIN) \
+	-e CNI_SPEC_VERSION=$(CNI_SPEC_VERSION) \
+	-e DATASTORE_TYPE=etcdv3 \
+	-e ETCD_ENDPOINTS=http://$(LOCAL_IP_ENV):2379 \
+	-v $(CURDIR):/go/src/$(PACKAGE_NAME):rw \
+	$(LOCAL_BUILD_MOUNTS) \
+	$(CALICO_BUILD) sh -c '\
+			cd  /go/src/$(PACKAGE_NAME) && \
+			ginkgo -cover -r -skipPackage vendor -skipPackage k8s-install $(GINKGO_ARGS)'
+	sudo mv ./report/azure_suite.xml ./report/azure_suite_etcd.xml
```

Comment:
I don't think these need `sudo` - we should remove any instances of `sudo` from the Makefile.

Reply:
Resolved in this commit: https://github.com/projectcalico/cni-plugin/pull/698/commits/67ece493d341032c9b39c0ee13824a3e680d1e39